addons/project_todo/models/res_users.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import json
+
+from odoo import _, api, fields, models, modules
+
+
+class Users(models.Model):
+    _name = 'res.users'
+    _inherit = ['res.users']
+
+    @api.model
+    def systray_get_activities(self):
+        """ Split To-do and Project activities in systray by removing
+            the single project.task activity represented and doing a
+            new query to split them between private/non-private tasks.
+        """
+        activity_groups = super(Users, self).systray_get_activities()
+        # 1. removing project.task activity group
+        for group in activity_groups:
+            if group.get('model') == 'project.task':
+                activity_groups.remove(group)
+                break
+
+        # 2. creating groups for todo and task seperately
+        query = """SELECT BOOL(t.project_id) as is_task, count(*), act.res_model as model, act.res_id,
+                       CASE
+                           WHEN %(today)s::date - act.date_deadline::date = 0 Then 'today'
+                           WHEN %(today)s::date - act.date_deadline::date > 0 Then 'overdue'
+                           WHEN %(today)s::date - act.date_deadline::date < 0 Then 'planned'
+                        END AS states
+                     FROM mail_activity AS act
+                     JOIN project_task AS t ON act.res_id = t.id
+                    WHERE act.res_model = 'project.task' AND act.user_id = %(user_id)s
+                 GROUP BY is_task, states, act.res_model, act.res_id;
+                """
+        self.env.cr.execute(query, {
+            'today': fields.Date.context_today(self),
+            'user_id': self.env.uid,
+        })
+        activity_data = self.env.cr.dictfetchall()
+
+        user_activities = {}
+        for act in activity_data:
+            if not user_activities.get(act['is_task']):
```

Comment:
```suggestion
            is_task = activity['is_task']
            if is_task not in user_activities:
```
- a variable as we'll use it a lot
- use in i/o .get bc the value can't be falsy anyway, right?